numba/cpython/cmathimpl.py

```diff
@@ -53,39 +54,32 @@ def isfinite_float_impl(context, builder, sig, args):
     return impl_ret_untracked(context, builder, sig.return_type, res)
 
 
-@lower(cmath.rect, types.Float, types.Float)
-def rect_impl(context, builder, sig, args):
-    [r, phi] = args
-    # We can't call math.isfinite() inside rect() below because it
-    # only exists on 3.2+.
-    phi_is_finite = mathimpl.is_finite(builder, phi)
-
-    def rect(r, phi, phi_is_finite):
-        if not phi_is_finite:
-            if not r:
-                # cmath.rect(0, phi={inf, nan}) = 0
-                return abs(r)
-            if math.isinf(r):
-                # cmath.rect(inf, phi={inf, nan}) = inf + j phi
-                return complex(r, phi)
-        real = math.cos(phi)
-        imag = math.sin(phi)
-        if real == 0. and math.isinf(r):
-            # 0 * inf would return NaN, we want to keep 0 but xor the sign
-            real /= r
-        else:
-            real *= r
-        if imag == 0. and math.isinf(r):
-            # ditto
-            imag /= r
-        else:
-            imag *= r
-        return complex(real, imag)
+@overload(cmath.rect)
+def impl_cmath_rect(r, phi):
+    if all([isinstance(typ, types.Float) for typ in [r, phi]]):
+        def impl(r, phi):
+            if not math.isfinite(phi):
```

Comment:
Are we sure this works in every case ? 

(Referring to comment ` # We can't call math.isfinite() inside rect() below because it only exists on 3.2+.`, if that is still valid. )

Reply:
From the docs, it says that Numba requires Python 3.7 or newer. See:
https://numba.readthedocs.io/en/stable/user/installing.html#compatibility